Add fpga-rsu role to install fpga packages

ratailor opened the pull request:
This change adds fpga-rsu ansible-role to install
OPAE FPGA packages on hosts where fpga hardware
is already installed.

ratailor committed:
Add fpga-rsu role to install fpga packages

This change adds fpga-rsu ansible-role to install
OPAE FPGA packages on hosts where fpga hardware
is already installed.

piotrkpc commented on defaults/main.yml:
```diff
@@ -0,0 +1,6 @@
+---
+# defaults file for fpga-rsu
+fpga_rsu_driver_pkg_url: ""
+fpga_rsu_tarball_name: "{{ tripleo_fpga_rsu_driver_pkg_url | basename }}"
```
IMHO templating should not be used in defaults, instead defaults should provide sensible defaults values and that will be overridden in vars or extra vars.

piotrkpc commented on tasks/install_fpga_packages.yml:
```diff
@@ -0,0 +1,143 @@
+    - name: check fpga driver package directory exist
+      stat:
+        path: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_dir_result
```
I don't think this part is needed since the file module should be idempotent.

piotrkpc commented on tasks/install_fpga_packages.yml:
```diff
@@ -0,0 +1,143 @@
+
+    - name: Fail if fpga driver tarball download fail.
+      fail:
+        msg: "FPGA Driver tarball download failed. Please check correct tarball URL is provided."
```
since this would help with debuging maybe adding the actual URL in the log msg could be helpful

piotrkpc commented on tasks/install_fpga_packages.yml:
```diff
@@ -0,0 +1,143 @@
+      register: fpga_driver_tarball_untar
+      failed_when: false
+
+- name: Setting fact and get rpm list
+  block:
+    - name: setting fact for fpga driver rpm directory
+      set_fact:
+        fpga_driver_rpm_dir: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name | splitext | first | splitext | first }}"
+
+    - name: get rpm list for installation
+      shell: ls -1 *.rpm
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_list_of_rpms
+      when: fpga_driver_tarball_untar is not failed
+
+    - name: get version of opae package
+      shell: rpm -qp --queryformat '%{VERSION}-%{RELEASE}' "opae-libs*"
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}/opae"
+      register: opae_package_version
+      when: opae_list_of_rpms.stdout_lines | length > 0
```
I'm not sure if this and corresponding check below are needed...

piotrkpc commented on tasks/install_fpga_packages.yml:
```diff
@@ -0,0 +1,143 @@
+      register: check_opae_packages_installed
+      when: opae_packages_installed.rc == 0
+
+    - name: Check installed packages version
+      shell:
+        rpm -q --queryformat '%{VERSION}-%{RELEASE}' "opae-libs"
+      register: opae_installed_version
+
+    - name: Fail when rpm package version and installed version mismatch.
+      fail:
+        msg: "Specified RPM package version and installed version mismatch."
+      when: opae_package_version.stdout != opae_installed_version.stdout
```
...those checks. IMHO `rpm --verify` is checking more and you don't need to register and handle new variables.

piotrkpc commented on tasks/install_fpga_packages.yml:
```diff
@@ -0,0 +1,143 @@
+  block:
+    - name: install fpga drivers
+      shell:
+        rpm -ivh {{ opae_list_of_rpms.stdout_lines | join(' ') }}
```
What will happen if rpms are already installed? Should we update rpms in this step (use `-U` flag)?

piotrkpc reviewed (changes requested):
I've marked some parts that I think might be worth improving.

ratailor commented:
> I've marked some parts that I think might be worth improving.

Thanks for the review. I have addressed your comments and will update the changes soon.
